Source/WebCore/dom/SlotAssignment.cpp

```diff
@@ -385,6 +386,9 @@ void NamedSlotAssignment::willRemoveAssignedNode(const Node& node, ShadowRoot&)
     slot->assignedNodes.removeFirstMatching([&node](const auto& item) {
         return item.get() == &node;
     });
+
+    if (auto* slotElement = slot->element.get())
+        Style::PseudoClassChangeInvalidation styleInvalidation(*slotElement, CSSSelector::PseudoClass::HasSlotted, !slot->assignedNodes.isEmpty());
```

Comment:
We need to create this object before we remove the matching items in the lines above and destroy it after we removed the items so something like this:
```
std::optional<Style::PseudoClassChangeInvalidation> styleInvalidation;
if (auto* slotElement = slot->element.get())
    styleInvalidation.emplace(*slotElement, CSSSelector::PseudoClass::HasSlotted, !slot->assignedNodes.isEmpty());
slot->assignedNodes.removeFirstMatching([&node](const auto& item) {
    return item.get() == &node;
});

```